oak_containers_orchestrator/src/crypto.rs

```diff
@@ -0,0 +1,101 @@
+//
+// Copyright 2023 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::proto::oak::containers::v1::{
+    orchestrator_crypto_server::OrchestratorCrypto, GetSessionKeysRequest, GetSessionKeysResponse,
+    GetSignatureRequest, GetSignatureResponse, KeyOrigin,
+};
+use oak_crypto::encryptor::{EncryptionKeyProvider, RecipientContextGenerator};
+use std::sync::Arc;
+use tonic::{Request, Response};
+
+pub struct KeyStore {
+    instance_encryption_key: EncryptionKeyProvider,
+    group_encryption_key: EncryptionKeyProvider,
+}
+
+impl KeyStore {
+    pub fn new() -> Self {
+        let instance_encryption_key = EncryptionKeyProvider::generate();
+        let group_encryption_key = instance_encryption_key.clone();
+        Self {
+            instance_encryption_key,
+            group_encryption_key,
+        }
+    }
+
+    // TODO(#4442): Currently we have to give the encryption key provider to the `ipc_server`.
+    // Once we move all enclave apps to the new crypto service - this function should be removed.
+    pub fn instance_encryption_key(&self) -> EncryptionKeyProvider {
```

Comment:
AFAICT this should be removed when we fix the Attester, not when we have a crypto service, so the TODO issue and comment is misleading.

nit: comments above a function declaration should be a doc comment. This should be moved inside.